src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -479,7 +479,7 @@ void QgsRelationReferenceWidget::init()
 
     QSet<QString> requestedAttrs;
 
-    QgsVectorLayerCache* layerCache = new QgsVectorLayerCache( mReferencedLayer, 100000, this );
+    QgsVectorLayerCache* layerCache = new QgsVectorLayerCache( mReferencedLayer, 200000, this );
```

Comment:
Should we go for a QSettings value here (with increased default)?

Reply:
I think it could be a good idea, having this hardcoded is a big trick.
 Even though it will be a very advanced user  setting. 
Sidenote, do we have a place to expose really advanced settings without cluttering the UI? like firefox has for instance with its "about:config" page ..

Reply:
In settings -> Options -> Advanced. By the way, the warning there was inspired by the firefox way of doing things ;)

Reply:
oh dear.. and I knew that already.. /me is Dory 
![image](https://user-images.githubusercontent.com/1629853/33932615-b4666fb8-dff3-11e7-8aba-65ab03ac07e5.png)

Reply:
> In settings -> Options -> Advanced.

@m-kuhn I added an entry for the relation reference widget. Let me know if you agree with that :)

![refrel](https://user-images.githubusercontent.com/9266424/33955895-63143812-e03d-11e7-9d72-526f4b370906.png)